### DISCUSSION TITLE
Auto-detect coverage bounding boxes for brain MRI images project

### DISCUSSION THREAD

**Discussion by Nouran-Muhammad** [Emory-HITI/Niffler#383] [General] [answered]
Hi all,
I am Nouran Muhammad, a final year student at systems and biomedical engineering department at Cairo university. I am interested in the project: Auto-detect coverage bounding boxes for brain MRI images. I have studied anatomy, image processing, DICOM images and deep learning at university and I have worked on projects where i used deep learning to segment tumors in MRI brain images. I worked also on brain MRI DICOM images.

I have installed Niffler and understood how to use it. I successfully ran the PNG extractor on a dataset of brain MRI DICOM images that i have previously installed from Kaggle. Do i have to download a dataset from TCIA and test on it as stated in the project description? since the brain MRI datasets are limited access and i have to sign a license and send it and wait for the access. 
Also, is there any prerequisite task to be done other than using Niffler with DICOM images?
Thanks

**Comment by pradeeban** [comment, marked as answer]
Hi and welcome, @Nouran-Muhammad,
"Do i have to download a dataset from TCIA and test on it as stated in the project description? since the brain MRI datasets are limited access and i have to sign a license and send it and wait for the access."

You can use the publicly available images (TCIA has some public images, too) for now.

There are no prerequisites other than those listed under the project idea.

**Reply by Nouran-Muhammad** [reply]
Thank you for your reply!
I do found some public images and extracted them to PNG, then i used opencv library to find contours in brain image (e.g. cerebral hemispheres, eyeballs...etc.)  and draw bounding boxes around them.
 
I have some questions. 
1- I noticed that the PNG extractor takes the DICOM images names and gets its sha224 and use it for PNG names. I am quite curious to know why do you do this? does this have a purpose or meaning?
2- Is it available to ask for the project's mentors' review and feedback about my proposal before submitting? and if it is available, how should i communicate with them?

**Reply by pradeeban** [reply]
Good observations.

1. The PNG extraction aims to make the images fully anonymized. We achieve that by removing any identifiable IDs. The instance names of the images are unique and can be linked to a patient. Therefore, to anonymize that, we make sha224 for its name.
2. You can email the project proposal as a Google Doc to the mentors. But please note we comment only if we have a clarification to make or if we need additional details.

**Reply by Nouran-Muhammad** [reply]
Thank you for your reply.
